packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -39,34 +39,34 @@ const createUpgradableConstructor = () => {
     class UpgradableConstructor extends HTMLElement {
         static formAssociated = true;
 
-        constructor(upgradeCallback: LifecycleCallback) {
+        constructor(upgradeCallback: LifecycleCallback, useNativeLifecycle: boolean) {
             super();
+
+            if (useNativeLifecycle) {
+                // When in native lifecycle mode, we need to keep track of instances that were created outside LWC
+                // (i.e. not created by `lwc.createElement()`). If the element uses synthetic lifecycle, then we don't
+                // need to track this.
+                nativeLifecycleElementsToUpgradedByLWC.set(this, elementBeingUpgradedByLWC);
```

Comment:
Nice!

Reply:
Yeah I was pleased that I was able to do this with just one `Map`!